lib/dma.c

```diff
@@ -137,6 +137,10 @@ dma_controller_remove_regions(lm_ctx_t *ctx, dma_controller_t *dma)
 void
 dma_controller_destroy(lm_ctx_t *lm_ctx, dma_controller_t *dma)
 {
+    if (dma == NULL) {
```

Comment:
I think the caller should not pass NULL *dma. It looks better to fix the caller and maybe have a assert() here.